src/servers/ZoneServer2016/managers/decaymanager.ts

```diff
@@ -64,6 +66,32 @@ export class DecayManager {
   private contructionExpirationCheck(server: ZoneServer2016) {
     for (const a in server._constructionFoundations) {
       const foundation = server._constructionFoundations[a];
+      if (
+        foundation.itemDefinitionId == Items.FOUNDATION ||
+        foundation.itemDefinitionId == Items.GROUND_TAMPER
+      ) {
+        if (
+          Date.now() - foundation.placementTime >=
+          this.griefFoundationTimer * 3600000
+        ) {
+          if (
+            Object.keys(foundation.occupiedWallSlots).length <
+              this.griefCheckSlotAmount &&
+            Object.keys(foundation.occupiedShelterSlots).length == 0 &&
+            Object.keys(foundation.occupiedExpansionSlots).length == 0
+          ) {
+            for (const a in foundation.occupiedWallSlots) {
+              foundation.occupiedWallSlots[a].destroy(server);
+            }
+            // clear floating entities
+            for (const a in foundation.freeplaceEntities) {
+              foundation.freeplaceEntities[a].destroy(server);
+            }
+            foundation.destroy(server);
```

Comment:
Maybe just add a log if a foundation is destroyed due to this system so we can see how often it's trigerred

Reply:
Currently it's 3 days, but i'll add the log for it 👍🏿

Reply:
Yep but more to see how many grief bases were deleted 🙂